### PR TITLE
CATROID-788 Add project options view

### PR DESCRIPTION
**Pull request by dandrawes**
https://jira.catrob.at/browse/CATROID-788
- added project options view in project overflow menu and on long press in the project list overview
- removed show details on long press in the project list overview since it is reachable through the project options view
- added testcase
- removed duplicate menu items
- show existing tags, but not selectable

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by ndrnour** [comment]
please rebase and fix the warning if still exists.

**Comment by ndrnour** [comment]
as it's mentioned in the ticket description, show details and upload menu items should be removed, did you forget that?

**Comment by ndrnour** [comment]
please have a look at the attached screen recording. and fix the menus shown in the More details view

https://user-images.githubusercontent.com/18437036/103477177-5890b100-4dbc-11eb-9d57-d1974c5d663c.mp4

**Review by ndrnour — CHANGES_REQUESTED** [review]
pls fix what's shown in the screen recording and implement it as requested

**Comment by dandrawes** [comment]
@ndrnour in the beginning it was planned to remove the duplicate menu items in a separate ticket, but now this is done here in a separate commit.

I also fixed the mentioned issue.

**Review by ndrnour — CHANGES_REQUESTED** [review]
please implement the changes as discussed over slack.

**Review by ndrnour — CHANGES_REQUESTED** [review]
one last thing, please squash your commits.

**Review by ndrnour — APPROVED** [review]

**Review by wslany — CHANGES_REQUESTED** [review]
Thanks! For some issues see inlined comments.

**Review by wslany — CHANGES_REQUESTED** [review]
When I 

1. download https://share.catrob.at/app/project/d8828099-cfd1-11ea-a5c3-005056a36f47 
2. go to project options
3. tap on the "More details" button

the app **_crashes_**. 

Stacktrace:
```
02-07 10:24:34.040 E/AndroidRuntime( 9591): FATAL EXCEPTION: main
02-07 10:24:34.040 E/AndroidRuntime( 9591): Process: org.catrobat.catroid.independent_20210207_0523, PID: 9591
02-07 10:24:34.040 E/AndroidRuntime( 9591): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.saveCreditsAndNotes(ProjectOptionsFragment.java:259)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.saveProject(ProjectOptionsFragment.java:209)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.onPause(ProjectOptionsFragment.java:201)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.Fragment.performPause(Fragment.java:2879)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentStateManager.pause(FragmentStateManager.java:383)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1206)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:442)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2169)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1992)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at android.os.Handler.handleCallback(Handler.java:883)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at android.os.Handler.dispatchMessage(Handler.java:100)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at android.os.Looper.loop(Looper.java:237)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at android.app.ActivityThread.main(ActivityThread.java:8154)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at java.lang.reflect.Method.invoke(Native Method)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): uncaughtException:
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.saveCreditsAndNotes(ProjectOptionsFragment.java:259)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.saveProject(ProjectOptionsFragment.java:209)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.onPause(ProjectOptionsFragment.java:201)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.Fragment.performPause(Fragment.java:2879)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentStateManager.pause(FragmentStateManager.java:383)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1206)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:442)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2169)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1992)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at android.os.Handler.handleCallback(Handler.java:883)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at android.os.Handler.dispatchMessage(Handler.java:100)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at android.os.Looper.loop(Looper.java:237)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at android.app.ActivityThread.main(ActivityThread.java:8154)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at java.lang.reflect.Method.invoke(Native Method)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
02-07 10:24:34.040 E/AndroidRuntime( 9591): FATAL EXCEPTION: main
02-07 10:24:34.040 E/AndroidRuntime( 9591): Process: org.catrobat.catroid.independent_20210207_0523, PID: 9591
02-07 10:24:34.040 E/AndroidRuntime( 9591): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.saveCreditsAndNotes(ProjectOptionsFragment.java:259)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.saveProject(ProjectOptionsFragment.java:209)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.onPause(ProjectOptionsFragment.java:201)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.Fragment.performPause(Fragment.java:2879)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentStateManager.pause(FragmentStateManager.java:383)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1206)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:442)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2169)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1992)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at android.os.Handler.handleCallback(Handler.java:883)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at android.os.Handler.dispatchMessage(Handler.java:100)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at android.os.Looper.loop(Looper.java:237)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at android.app.ActivityThread.main(ActivityThread.java:8154)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at java.lang.reflect.Method.invoke(Native Method)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
02-07 10:24:34.040 E/AndroidRuntime( 9591): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): uncaughtException:
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.saveCreditsAndNotes(ProjectOptionsFragment.java:259)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.saveProject(ProjectOptionsFragment.java:209)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at org.catrobat.catroid.ui.fragment.ProjectOptionsFragment.onPause(ProjectOptionsFragment.java:201)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.Fragment.performPause(Fragment.java:2879)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentStateManager.pause(FragmentStateManager.java:383)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1206)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:442)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2169)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1992)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at android.os.Handler.handleCallback(Handler.java:883)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at android.os.Handler.dispatchMessage(Handler.java:100)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at android.os.Looper.loop(Looper.java:237)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at android.app.ActivityThread.main(ActivityThread.java:8154)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at java.lang.reflect.Method.invoke(Native Method)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
02-07 10:24:34.041 E/BaseExceptionHandler( 9591): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
02-07 10:24:34.120 E/Zygote  ( 4162): isWhitelistProcess - Process is Whitelisted
02-07 10:24:34.121 E/Zygote  ( 4162): accessInfo : 1
02-07 10:24:34.139 E/t_20210207_052( 4162): Unknown bits set in runtime_flags: 0x8000
02-07 10:24:34.120 E/Zygote  ( 4162): isWhitelistProcess - Process is Whitelisted
02-07 10:24:34.121 E/Zygote  ( 4162): accessInfo : 1
02-07 10:24:34.139 E/t_20210207_052( 4162): Unknown bits set in runtime_flags: 0x8000

Switch Device

Local testing
-
+
Zoom

DevTools

Report a bug

Settings

Stop Session

```

File: catroid/src/main/java/org/catrobat/catroid/content/XmlHeader.java [hunk 1]
```diff
@@ -242,6 +243,10 @@ public boolean isCastProject() {
 		return isCastProject;
 	}
 
+	public List<String> getTags() {
+		return Arrays.asList(this.tags.split(","));
+	}
+
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/content/XmlHeader.java:249]: please remove _@SuppressWarnings("unused")_ from line 68

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/ProjectOptionsFragment.java [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+		BottomBar.hideBottomBar(getActivity());
+		return view;
+	}
+
+	private void addTags() {
+		tagsChipGroup.removeAllViews();
+		tags = project.getTags();
+		LinearLayout tagsLayout = view.findViewById(R.id.tags);
+
+		if (tags.size() == 1 && tags.get(0).isEmpty()) {
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/ProjectOptionsFragment.java:156]:
```suggestion
		if (tags.size() > 0 && tags.get(0).isEmpty()) {
```

**Review by ndrnour — APPROVED** [review]
LGTM

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!

**Pull request merged**